go-controller/pkg/kube/annotator_test.go

```diff
@@ -0,0 +1,605 @@
+package kube
+
+import (
+	"fmt"
+	ovntest "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/testing"
+	mock_clientgo_iface "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/testing/mocks/k8s.io/client-go/kubernetes"
+	mock_core_v1 "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/testing/mocks/k8s.io/client-go/kubernetes/typed/core/v1"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	v1 "k8s.io/api/core/v1"
+	"testing"
+)
+
+func TestNewNodeAnnotator(t *testing.T) {
+	node := &v1.Node{}
+	//Kube defined in kube.go
+	k := &Kube{}
+	tests := []struct {
+		desc string
+	}{
+		{
+			desc: "Positive test code path, returns a new node annotator",
+		},
+	}
+	for i, tc := range tests {
+		t.Run(fmt.Sprintf("%d:%s", i, tc.desc), func(t *testing.T) {
+
+			e := NewNodeAnnotator(k, node)
+
+			assert.NotNil(t, e)
+		})
+	}
+}
+
+func TestNodeAnnotator_Set(t *testing.T) {
+	nodeAnnotator := nodeAnnotator{}
+	nodeAnnotator.changes = make(map[string]*action)
+	tests := []struct {
+		desc string
+		val  interface{}
+	}{
+		{
+			desc: "Positive test code path",
+			val:  "annotations",
+		},
+	}
+	for i, tc := range tests {
+		t.Run(fmt.Sprintf("%d:%s", i, tc.desc), func(t *testing.T) {
+
+			e := nodeAnnotator.Set("key", tc.val)
+
+			assert.Nil(t, e)
+		})
+	}
+}
+
+func TestNodeAnnotator_SetWithFailureHandler(t *testing.T) {
+	nodeAnnotator := nodeAnnotator{}
+	nodeAnnotator.changes = make(map[string]*action)
+	tests := []struct {
+		desc string
+		val  interface{}
+	}{
+		{
+			desc: "Test code path when annotations is Nil",
+			val:  nil,
+		},
+		{
+			desc: "Test code path; annotations is not of type string",
+			val:  map[string]string{},
+		},
+		{
+			desc: "Test code path; annotations is string",
+			val:  "annotations",
+		},
+	}
+	for i, tc := range tests {
+		t.Run(fmt.Sprintf("%d:%s", i, tc.desc), func(t *testing.T) {
+
+			e := nodeAnnotator.SetWithFailureHandler("key", tc.val, nil)
+
+			assert.Nil(t, e)
+		})
+	}
+}
+
+func TestNodeAnnotator_Delete(t *testing.T) {
+	nodeAnnotator := nodeAnnotator{}
+	nodeAnnotator.changes = make(map[string]*action)
+	tests := []struct {
+		desc string
+		key  string
+	}{
+		{
+			desc: "Positive test code path",
+			key:  "key",
+		},
+	}
+	for i, tc := range tests {
+		t.Run(fmt.Sprintf("%d:%s", i, tc.desc), func(t *testing.T) {
+
+			nodeAnnotator.Delete(tc.key)
+
+		})
+	}
+}
+
+func TestNodeAnnotator_Run(t *testing.T) {
+	mockIface := new(mock_clientgo_iface.Interface)
+	mockCoreIface := new(mock_core_v1.CoreV1Interface)
+	mockNodeIface := new(mock_core_v1.NodeInterface)
+	nodeAnnotator := nodeAnnotator{}
+	nodeAnnotator.kube = &Kube{KClient: mockIface}
+	nodeAnnotator.changes = make(map[string]*action)
+	nodeAnnotator.node = &v1.Node{}
+	nodeAnnotator.node.Annotations = make(map[string]string)
+	nodeAnnotator.node.Annotations["key"] = ""
+	tests := []struct {
+		desc                   string
+		expectedErr            error
```

Comment:
if you are not testing for the return of specific errors expectedErr should be a boolean

Reply:
True, @sheetalsingala We may not be consistent everywhere but have tried to use expectedErr or errAssert in most of the places, I believe, please refer https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/util/net_unit_test.go#L70-L71